packages/protocol/contracts/governance/LockedGold.sol

```diff
@@ -236,4 +246,73 @@ contract LockedGold is ILockedGold, ReentrancyGuard, Initializable, UsingRegistr
     list[index] = list[lastIndex];
     list.length = lastIndex;
   }
+
+  /**
+   * @notice Adds `slasher` to whitelist of approved slashing addresses.
+   * @param slasher Address to whitelist.
+   */
+  function addSlasher(address slasher) external onlyOwner {
+    require(slasher != address(0));
+    isSlasher[slasher] = true;
+    emit SlasherWhitelistAdded(slasher);
+  }
+
+  /**
+   * @notice Removes `slasher` from whitelist of approved slashing addresses.
+   * @param slasher Address to remove from whitelist.
+   */
+  function removeSlasher(address slasher) external onlyOwner {
+    require(isSlasher[slasher]);
+    isSlasher[slasher] = false;
+    emit SlasherWhitelistRemoved(slasher);
+  }
+
+  /**
+   * @notice Slashes `account` by reducing its nonvoting locked gold by `penalty`.
+   *         If there is not enough nonvoting locked gold to slash, calls into
+   *         `Election.slashVotes` to slash the remaining gold. Also sends `reward`
+   *         gold to the reporter, and penalty-reward to the Community Fund.
+   * @param account Address of account being slashed.
+   * @param penalty Amount to slash account.
+   * @param reporter Address of account reporting the slasher.
+   * @param reward Reward to give reporter.
+   * @param lessers The groups receiving fewer votes than i'th group, or 0 if the i'th group has
+   *                the fewest votes of any validator group.
+   * @param greaters The groups receiving more votes than the i'th group, or 0 if the i'th group
+   *                 has the most votes of any validator group.
+   * @param indices The indices of the i'th group in `account`'s voting list.
+   */
+  function slash(
+    address account,
+    uint256 penalty,
+    address reporter,
+    uint256 reward,
+    address[] calldata lessers,
+    address[] calldata greaters,
+    uint256[] calldata indices
+  ) external onlySlasher {
+    require(
+      getAccountTotalLockedGold(account) >= penalty,
+      "trying to slash more gold than is locked"
+    );
+    require(penalty >= reward, "reward cannot exceed penalty.");
+    {
+      uint256 nonvotingBalance = balances[account].nonvoting;
+      uint256 difference = 0;
+      // If not enough nonvoting, revoke the difference
+      if (nonvotingBalance < penalty) {
+        difference = penalty.sub(nonvotingBalance);
+        require(
```

Comment:
What should we do if there's not enough to slash anymore?

Reply:
I believe we would want to just slash all of the `account`'s gold, I'm assuming this is handled on the caller side (`Slasher` contract) by passing in for `penalty` `max(normalSlashPenalty, totalGoldOwnedByAccount)` where `normalSlashPenalty` is the default penalty and `totalGoldOwnedByAccount` is the total balance + voting balance for the account.

Reply:
Now that the other PRs are in, we can update e.g. https://github.com/celo-org/celo-monorepo/blob/master/packages/protocol/contracts/governance/GovernanceSlasher.sol#L64 to do the ` max(normalSlashPenalty, totalGoldOwnedByAccount)`.

Reply:
Perhaps it's easier to handle in LockedGold, because the needed balances are stored there.

Reply:
Based on this I am changing LockedGold.slash to only slash min(totalLockedGold, penalty). A problematic implication is that if totalLockedGold is somehow less than the reward passed in to slash, the call will revert. However, without doing a bunch of wasteful calculations (adjusting reward based on its original proportion or something similar) slash doesn't have the right info to change rewardproperly. Just something to keep in mind

Reply:
I think this is OK for now, but i would follow the discussion on this. 

My concern is that Given the `min()` i will be considered "slashed" and thus can't be slashed again for the same thing, but i wouldn't be really paying the penalty that i'm due... i wonder if i can use that to shield me from penalties somehow.

But to analyze this we need to see the bigger picture...